### PR TITLE
fix lags in WhoReacted plugin

### PR DESCRIPTION
**Pull request by Juby210**

### CODE REVIEW

File: WhoReacted/WhoReacted.plugin.js [hunk 1]
```diff
@@ -245,29 +251,15 @@ module.exports = !global.ZeresPluginLibrary ? class {
...
             );
 
-            Reactions.forceUpdateAll();
+            ReactComponents.getComponentByName("Reactions", DiscordSelectors.Reactions.reactions).then(c => c.forceUpdateAll());
         }
 
         async findReaction() {
-            const Reactions = await this.findReactions();
-
-            const unpatch = Patcher.after(Reactions.component.prototype, "render", (thisObject, args, returnValue) => {
```
Comment (jaimeadf) [inline WhoReacted/WhoReacted.plugin.js:254]: Sometimes the Reaction component isn't found by just calling `ReactComponents.getComponentByName` usually after you have just opened discord. When you push the component it is put in a cache and then no more problems will happen when finding it.

File: WhoReacted/WhoReacted.plugin.js [hunk 2]
```diff
@@ -156,34 +159,48 @@ module.exports = !global.ZeresPluginLibrary ? class {
             const VoiceUserSummaryItemComponent = WebpackModules.find(m => m.default?.displayName === "VoiceUserSummaryItem").default;
 
             const Reaction = await this.findReaction();
-            const Reactions = await this.findReactions();
+            const Reactions = this.findReactions();
```
Comment (jaimeadf) [inline WhoReacted/WhoReacted.plugin.js:162]: I think it's better to find the Reactions using the ReactComponent util to reuse again below when forcing update.

**Comment by jaimeadf** [comment]
I like the other improvements and I didn't know that you can just replace the function on the Reaction render.

**Comment by Juby210** [comment]
After your commit lag presents again but is smaller.

image

**Comment by jaimeadf** [comment]
I could be wrong but I think the little bigger lag came from the readdition of the patch to find the Reaction component which usually isn't found without that patch.

**Pull request merged**